geopandas/plotting.py

```diff
@@ -582,10 +582,21 @@ def plot_dataframe(
         # set categorical to True for creating the legend
         categorical = True
         binedges = [values[~nan_idx].min()] + binning.bins.tolist()
-        categories = [
-            "{0:.2f} - {1:.2f}".format(binedges[i], binedges[i + 1])
-            for i in range(len(binedges) - 1)
-        ]
+        if legend_kwds is not None and 'labels' in legend_kwds:
+            if len(legend_kwds['labels']) != classification_kwds['k']:
+                raise AttributeError('Number of labels must match number of bins, '
+                                     'received {} labels for {} bins'.
+                                     format(len(legend_kwds['labels']),
+                                            classification_kwds['k']))
+            else:
+                categories = list(legend_kwds['labels'])
+                # The 'label' keyword raises an error
```

Comment:
Which error? We might be duplicating some maptlotlib keywords here. In that case we should avoid it.

Reply:
I had an error because I was using 'label' and not 'labels'. Now it doesn't throw any error but if I don't delete the 'labels' key I get : 
`UserWarning: You have mixed positional and keyword arguments, some input may be discarded.
  ax.legend(patches, categories, **legend_kwds)` and no legend is shown. You are right, maybe another keyword might be better, sth like `bin_labels` ?

Reply:
In `ax.legend(patches, categories, **legend_kwds)`  categories are, in fact, `labels`. So we can keep it because that is what you are passing.